services/placement/service/placementservice.go

```diff
@@ -118,40 +129,56 @@ func (ps placementService) AddInstance(
 		return nil, err
 	}
 
-	return p, ps.validateAndSaveSnapshot(p)
+	if err := placement.Validate(p); err != nil {
+		return nil, err
+	}
+
+	// TODO(chaowant) this will be removed once the m3db nodes start to mark shards as available
+	p, err = markAllShardsAsAvailable(p)
```

Comment:
Yeah we should not mark these shards as available, we need to rely on nodes marking them when they are initialized.